algebird-core/src/main/scala/com/twitter/algebird/Max.scala

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2016 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package com.twitter.algebird
+
+import scala.annotation.tailrec
+
+// To use the MaxSemigroup wrap your item in Max
+case class Max[@specialized(Int, Long, Float, Double) +T](get: T) {
+  def max[U >: T](r: Max[U])(implicit ord: Ordering[U]): Max[U] =
+    Max.ordering.max(this, r)
+  def +[U >: T](r: Max[U])(implicit ord: Ordering[U]): Max[U] = max(r)
+}
+
+object Max extends MaxInstances {
+  def aggregator[T](implicit ord: Ordering[T]): MaxAggregator[T] = MaxAggregator()(ord)
+}
+
+private[algebird] sealed abstract class MaxInstances {
+  implicit def equiv[T](implicit eq: Equiv[T]): Equiv[Max[T]] = Equiv.by(_.get)
+
+  implicit def semigroup[T](implicit ord: Ordering[T]): Semigroup[Max[T]] =
+    Semigroup.from[Max[T]] { (l, r) => if (ord.gteq(l.get, r.get)) l else r }
+
+  implicit def ordering[T](implicit ord: Ordering[T]): Ordering[Max[T]] =
+    Ordering.by(_.get)
+
+  // Zero should have the property that it <= all T
+  def monoid[T](zero: => T)(implicit ord: Ordering[T]): Monoid[Max[T]] =
+    Monoid.from(Max(zero)) { (l, r) => if (ord.gteq(l.get, r.get)) l else r }
```

Comment:
Would be nice to override sumOption while we are at it can avoid a lot of reboxing.